### DISCUSSION TITLE
Linker for cxsnippets

### DISCUSSION THREAD

**Discussion by jj683** [xmake-io/xmake#5974] [Q&A]
Which linker should be used in `check_cxsnippet` if you using custom toolchain?
For example, with custom clang-18 toolchain with enabled LTO for test snippet of Catch2 library right now xmake using "ld" linker, which is different from "ld" env variable.

Debug log: https://gist.github.com/jj683/999b2705b3c0c36fec50d3b39751debc

**Comment by waruqi** [comment]
I don't know why just linking the test would enable lto, I have no more information.
